### DISCUSSION TITLE
How to cancel user creation?

### DISCUSSION THREAD

**Discussion by sachinraja** [nextauthjs/next-auth#2264] [Help] [answered]
In the `profile` callback, I am checking `localStorage` to see if `username` exists. Is there any way to cancel the creation of a new user without erroring if `username` does not exist?

**Comment by balazsorban44** [comment, marked as answer]
Please provide more information. Why would you save anything in `localStorage` about the user? Please note that this might be a security issue, as `localStorage` is easily readable by malicious actors.

Anyways, to stop a user being logged in, we provide the following API:

https://next-auth.js.org/configuration/callbacks#sign-in-callback

**Reply by sachinraja** [reply]
I am not saving anything sensitive about the user. Before each user signs up, I want them to create a unique username. My plan is to persist their requested username in `localStorage` and fetch that upon user creation.

**Reply by sachinraja** [reply]
Hey, thank you for letting me know. I plan on using a cookie (as `localStorage` cannot be accessed from an API anyway). I actually saw the `signIn` callback when I was researching earlier, but didn't think it was what I needed because it says you can use it to "control if a user is allowed to sign in." Perhaps it should be documented that it will actually cancel user creation if they do not already have an account?
